### PR TITLE
SONARJAVA-3663 S2976 implementation moved to S5445

### PR DESCRIPTION
**Pull request by margarita-nedzelska-sonarsource**

### CODE REVIEW

**Review by quentin-jaquier-sonarsource — CHANGES_REQUESTED** [review]
I think it makes sense to deprecate S2976 in this PR. Running ruleapi will not work, because the replacement rule is not yet merged. You can work around this by manually setting java in the "covered language" of the S5445.

In addition, this rule is now a "Blocker" (S2976 was "critical"). I feel that keeping the original severity for the new rule makes more sense.

**Comment by margarita-nedzelska-sonarsource** [comment]
I asked AppSec about the severity, they're going to think about it next week, so asked to keep it as is for now.
We'll still be able to adjust changes before the release

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/java/Sonar_way_profile.json [hunk 1]
```diff
@@ -380,6 +380,7 @@
     "S5411",
     "S5413",
     "S5443",
+    "S5445",
```
Comment (quentin-jaquier-sonarsource) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/java/Sonar_way_profile.json:383]: I would expect to see S2976 removed from the SonarWay profile, is it expected to keep it?

Comment (margarita-nedzelska-sonarsource) [reply java-checks/src/main/resources/org/sonar/l10n/java/rules/java/Sonar_way_profile.json:383]: I asked Eric, he said, it should be done automatically, once we deprecate it.
I don't understand exactly what is the process here. rule-api, didn't remove it, when I run scripts.

I went through this[ doc ](https://xtranet-sonarsource.atlassian.net/wiki/spaces/LANG/pages/665976/Coding+Rules+Guidelines) and it looks like we need to update RSPEC itself and remove it from SonarWay, but I am not absolutely sure what's done with script and what should be done manualy.

We can also wait till tomorrow to see if any jobs did some updates on the rule.

Comment (quentin-jaquier-sonarsource) [reply java-checks/src/main/resources/org/sonar/l10n/java/rules/java/Sonar_way_profile.json:383]: Indeed, "sonartech" did some changes to the RSPEC this night, removing the default profile.

Comment (margarita-nedzelska-sonarsource) [reply java-checks/src/main/resources/org/sonar/l10n/java/rules/java/Sonar_way_profile.json:382]: Updated PR :)

**Review by quentin-jaquier-sonarsource — APPROVED** [review]

**Pull request merged**